elytron/ELY-201-Add-client-side-HTTP-authentication-mechanism-support.adoc

```diff
@@ -0,0 +1,171 @@
+
+We will create a class called ElytronHttpClient which contains method connect().
+
+We will create an object of HttpClient in the constructor of ElytronHttpClient.
+
+[source,xml]
+----
+public ElytronHttpClient(){
+   client = HttpClient.newHttpClient();
+}
+----
+
+connect() method requires a parameter uri which is of type String,In this method we first get an instance of HttpRequest using buildRequest() method and using HttpClient object we will send the request by send() method which return HttpResponse.
+
+public HttpResponse<String> connect(String uri,String method, String body, Map<String, String> headers)
+
+
+-> In case of Basic Authentication mechanism,we require username and password to authenticate the user for the given uri.For that we will add user credential to the wildfly-config.xml as following :-
+
+[source,xml]
+----
+<set-user-name name="quickstartUser"/>
+<credentials>
+   <clear-password password="quickstartPwd1!"/>
+</credentials>
+----
+
+We will use different type of authentication mechanism 
+Basic,bearer,certificate,digest,external
+
+Based on the mechanism type we will call different methods to get authorization value which will be added to the header for authentication.
+Like for basic type authentication mechanism we will create a class which contains evaluateMechanism() method :  
+
+[source,xml]
+----
+public static HttpRequest evaluateMechanism(URI uri) {
+    HttpRequest request = HttpRequest
+            .newBuilder()
+            .uri(uri)
+            .header(ElytronHttpClientConstants.AUTHORIZATION, basicAuth(userName, password))
+            .build();
+    return request;
+}
+----
+
+Similar to this we will have different classes for different authentication mechanism like digest,bearer,etc.
+
+Now after sending the request we will get a response which is of type HttpResponse.we will determine using the status if the authentication success or failed.If status code is 200 then we will know that authentication is successful.
+
+If the authentication failure then headers will contain : 
+www-authenticate=[Basic realm="RealmUsersRoles"] for basic authentication.similar for the other authentication mechanism type like digest,bearer,etc.
+
+
+== Security Considerations
+
+////
+Identification if any security implications that may need to be considered with this feature
+or a confirmation that there are no security implications to consider.
+////
+
+== Test Plan
+
+To test the above functionality we can add the tests as follows : 
+Inside tests/base/src/test/java/org/wildfly/security/http/client/ we will create a test class named as ElytronHttpClientTest where we will get an instance of HttpRequest from ElytronHttpClient#evaluateMechanism() method of different authentication mechanism and then we will create an object of TestingHttpServerRequest using request header which we will pass to the evaluateRequest() method of BasicAuthenticationMechanism which is used to evaluate our header values of request method.
```

Comment:
I am having difficulties understanding this long sentence.  Can you please reformulate this? Also by object of TestingHttpServerRequest you mean `instance` of right? since the class already exists. Also it is not clear when you are refering to the authentication mechanism of the client and when you are refering to the authentication mechanism of the server